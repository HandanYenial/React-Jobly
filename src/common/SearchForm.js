import React from 'react';
import { useState } from "react";
import "./SearchForm.css";

/**1. Show a search form with a text input and a submit button.
 * 2. It will appera both in company page and jobs page
 * 3. 
 * 
 * 
 */

function SearchForm({ searchFor }) { 
    console.debug("SearchForm" , "searchFor=" , typeof searchForm);

    const [searchTerm , setSearchTerm] = useState("");

    //1. Handle form submit: tell parent to filter
    //The trim() method removes whitespace from both ends of a string and returns a new string,
    // without modifying the original string. 
    function handleSubmit(evt){
        evt.preventDefault();
        searchFor(searchTerm.trim() || undefined);//take care of empty search term
        setSearchTerm(searchTerm.trim());
    }

    //2. Handle typing in search box
    function handleChange(evt){
        setSearchTerm(evt.target.value);
    }

  return (
    <div className = "SearchForm">
        <form className = "form-inline" onSubmit = {handleSubmit}>
            <input
                  className = "form-control form-control-lg flex-grow-1"
                  name = "searchTerm"
                  placeholder = "Enter search term ..."
                  value = { searchTerm }
                  onChange = {handleChange}
            />
            <button type = "submit"
                    className = "btn btn-lg btn-primary">
                        Search
            </button>
        </form>
    </div>
  );
}

export default SearchForm;