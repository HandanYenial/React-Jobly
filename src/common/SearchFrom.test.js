import React from "react";
import SearchForm from "./SearchForm";
import { render } from "@testing-library/react";

//Smoke test
it("renders without crashing", function(){
    render(<SearchForm/>);
});

//Snapshot test
it("matches with the snapshot" , function(){
    const { asFragment } = render(<SearchForm/>);
    expect(asFragment()).toMatchSnapshot();

});
