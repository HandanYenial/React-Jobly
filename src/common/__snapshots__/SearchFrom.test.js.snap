// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches with the snapshot 1`] = `
<DocumentFragment>
  <div
    class="SearchForm"
  >
    <form
      class="form-inline"
    >
      <input
        class="form-control form-control-lg flex-grow-1"
        name="searchTerm"
        placeholder="Enter search term ..."
        value=""
      />
      <button
        class="btn btn-lg btn-primary"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
